### PR TITLE
V2 : general cleanup of readme files

### PR DESCRIPTION
**Pull request by z3dev**
These are general changes to the README documents for each package.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

### CODE REVIEW

**Comment by z3dev** [comment]
TO DO:
- [ ] add README to @jscad/array-utils
- [ ] add missing sections in README files for:
- [ ] removed unrequired sections in README files for:

**Comment by z3dev** [comment]
During the updates, I'd like to make some adjustments to the README files.

For the UI modules, these will have full information:
- Usage
- Development
- Documentation
- Known issues
- Contribute
- Community
- Acknowledgements
- License
- See also

For component modules, these will have limited information:
- Usage
- Documentation ??
- Community
- Contribute
- Acknowledgements (if any apply)
- License

**Comment by z3dev** [comment]
@kaosat-dev what do you think about the above?

one question... should modeling have LICENSE? CONTRIBUTING?

**Comment by z3dev** [comment]
> I do have a few notes, and mostly a thought on how to save us maintenance work : perhaps the 'contributing' document should be moved to the root of the repo, and slightly changed so that it is valid for ALL UI packages (without the need of have 99% of the same text for all of them, in each folder), any specifics can be put into the README so there is only one main manual document to maintain for each .
> What do you think ?

Sounds good to me. Less documents to maintain is better.

I’ll merge the contributing documents that have now, and update the links in the package specific readme documents.

I think just one LICENSE is also best. Again, one document is easily maintained. And all the readme documents just link to the main one.

File: README.md [hunk 1]
```diff
@@ -93,30 +91,32 @@ Then, for example for CSG.js:
 
 You can now make changes to the CSG.js code and see it impact your locally running
 copy of OpenJSCAD live.
+
 ## Development
 
-Please see the code and details in each [package](./packages) & their READMEs
-You can link up all the code for simple development using the following command
+Please see the code and details in each [module](./packages) and their READMEs.
```
[the remark this answers is not in the document]
Comment (z3dev) [reply README.md:97]: So... just to get this correct...
- this project is composed of a set of PACKAGES
- which get published as NPM PACKAGES

So, 'module' is the wrong term, period. Correct?

File: README.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+>*OpenJsCad* is a set of modular, browser and command line tools for creating parametric 2D and 3D designs with JavaScript code.
 
 <img src="docs/logo.png" width="140" align="right" alt="OpenJSCAD">
 
 There are many ways to use OpenJSCAD: 
 
 An [online version](https://openjscad.org/), [self hosteable web based ui](https://github.com/jscad/OpenJSCAD.org/raw/master/packages/web), as [CLI](https://github.com/jscad/OpenJSCAD.org/raw/master/packages/cli) (command-line interface) for server-side computations with Node.js, as well as an experimental [desktop app](https://github.com/jscad/jscad-desktop) or [individual Node.js modules](https://github.com/jscad/OpenJSCAD.org/raw/master/packages/README.md)!!
 
-This repository is a [monorepo](https://medium.com/@maoberlehner/monorepos-in-the-wild-33c6eb246cb9) (container of multiple node.js packages & tools) maintaned with [Lerna](https://lernajs.io/)
+This repository is a [monorepo](https://medium.com/@maoberlehner/monorepos-in-the-wild-33c6eb246cb9) (container of multiple node.js modules and tools) maintained with [Lerna](https://lernajs.io/)
```
[the remark this answers is not in the document]
Comment (z3dev) [reply README.md]: Done

File: README.md [hunk 3]
```diff
@@ -128,36 +128,34 @@ Thank you to all the people who have already contributed to this project:
...
+* If you want to submit a change or a patch, please see the [Contributing guidelines](https://github.com/jscad/OpenJSCAD.org/blob/master/CONTRIBUTING.md). New contributions are accepted as [Pull Requests](https://github.com/jscad/OpenJSCAD.org/pulls/) via GitHub.
 
 * We only accept bug reports and pull requests on **GitHub**.
 
 * If you have a question about how to use the OpenJSCAD.org, then please start a conversation at the [OpenJSCAD.org User Group](https://jscad.xyz/forum). You might find the answer in the [OpenJSCAD.org User Guide](https://openjscad.org/dokuwiki/doku.php?id=start).
 
 * If you have a change or new feature in mind, please start a conversation with the [Core Developers](https://jscad.xyz/forum) and start contributing changes.
 
-Small Note: If editing this README, please conform to the [standard-readme](https://github.com/RichardLitt/standard-readme) specification.
+Small Note: If editing this README, please conform to the [standard readme](https://github.com/RichardLitt/standard-readme) specification.
```
[the remark this answers is not in the document]
Comment (z3dev) [reply README.md]: Done

**Comment by z3dev** [comment]
So... just to get this correct... as I think some of the documentation needs some changes.

- this project is composed of a set of PACKAGES
- which get published as NPM PACKAGES
- users can download and install the NPM PACKAGES as part of projects
- programs access and use the exported MODULES within each PACKAGE

**Comment by kaosat-dev** [comment]
@z3dev yes, that is exactly it !

**Comment by z3dev** [comment]
@kaosat-dev please review again. it's probably best to look at the results, not the differences.

this is taking too much time, and will probably never stop.
but if you want any more changes then let me know.

File: README.md [hunk 4]
```diff
@@ -106,11 +101,11 @@ HUGE THANKS and SHOUTOUT to [nodeBB](https://nodebb.org/) for hosting our (aweso
 
 JSCAD and all sub components are built upon great open source packages, and contributions.
 
-- [CSG Library](https://github.com/jscad/csg.js) by:
-Evan Wallace,
-Eduard Bespalov,
-Joost Nieuwenhuijse,
-Alexandre Girard
+Eary CSG Library by:
```
Comment (kaosat-dev) [inline README.md:104]: Small typo :)

File: README.md [hunk 5]
```diff
@@ -145,8 +138,6 @@ Your logo will show up here with a link to your website.
 <a href="https://opencollective.com/openjscad/sponsor/8/website" target="_blank"><img src="https://opencollective.com/openjscad/sponsor/8/avatar.svg"></a>
 <a href="https://opencollective.com/openjscad/sponsor/9/website" target="_blank"><img src="https://opencollective.com/openjscad/sponsor/9/avatar.svg"></a>
 
-HUGE THANKS and SHOUTOUT to [nodeBB](https://nodebb.org/) for hosting our (awesome) forum for free ! Thanks a lot !
```
Comment (kaosat-dev) [inline README.md]: Why remove this ?

File: README.md [hunk 6]
```diff
@@ -179,4 +170,4 @@ Drag and drop multiple files (Chrome and Firefox) or a folder (Chrome):
 
 That's all for now,
 
-Rene K. Mueller, Jeff Gay, Mark Moissette and JSCAD Organization
```
Comment (kaosat-dev) [inline README.md]: No more spotlight, fame & fortune for us ? :)

File: packages/cli/README.md [hunk 7]
```diff
@@ -104,10 +105,10 @@ See examples/module-design for such a design
 
 - or run the CLI on the main file if you know which one it is
 
-```openjscad <PATH TO DESIGN FOLDER>/index.js <OPTIONS>```
+```openjscad <PATH TO DESIGN FOLDER>/main.js <OPTIONS>```
```
Comment (kaosat-dev) [inline packages/cli/README.md:108]: why the change to main ? we want to make people have good practices, and we are in js, so entry points are usually index.js (just to be clear, it may seem like I am nitpicky, but I see no reason to change something that is not wrong)

File: packages/cli/README.md [hunk 8]
```diff
@@ -20,15 +20,16 @@ This is the Command Line Interface (CLI) package for creating designs using [Nod
...
+- [Sponsors](#sponsors)
 - [License](#license)
 
 ## Usage
 
 ### Install Node.js
 
 > IMPORTANT: You need a recent, LTS version of [Node.js](http://nodejs.org/)
-> Click on build "passing" above to understand which versions of Node.js are supported. Other versions of Node.js are not guaranteed to work !
+> Other versions of Node.js are not guaranteed to work!
```
Comment (kaosat-dev) [inline packages/cli/README.md:32]: Hmm Sorry to be a pain, but this provides a lot less information ?
Perhaps we can just link to the versions, or say clearly which versions are supported, but this is less informative as it is

File: README.md (hunk 5 above)
Comment (z3dev) [reply README.md]: I think this was repeated several times, so I removed the extra inserts. I’ll double check.

File: README.md (hunk 6 above)
Comment (z3dev) [reply README.md]: Well... Rene doesn’t really take part... and I’m shy. ☺️ 

I’d prefer to have sponsors and contributors. 😃

File: packages/cli/README.md (hunk 7 above)
Comment (z3dev) [reply packages/cli/README.md:108]: I really don’t know what’s correct. I’d be glad to add the hierarchy.

File: packages/cli/README.md (hunk 8 above)
Comment (z3dev) [reply packages/cli/README.md:32]: 😃 I added that before, and removed it. Ok. I’ll add that back.

File: README.md (hunk 4 above)
Comment (z3dev) [reply README.md]: Fixed

File: README.md (hunk 5 above)
Comment (z3dev) [reply README.md]: Contributors was added back.

File: packages/cli/README.md (hunk 7 above)
Comment (z3dev) [reply packages/cli/README.md]: Reverted. Now says index.js

File: packages/cli/README.md (hunk 8 above)
Comment (z3dev) [reply packages/cli/README.md:33]: I added this back.

**Comment by z3dev** [comment]
round one of small changes completed. :)

@kaosat-dev back to you!
